libbeat/beat/beat.go

```diff
@@ -303,3 +317,20 @@ func handleError(err error) error {
 func GetDefaultVersion() string {
 	return defaultBeatVersion
 }
+
+func startCollectingMetrics() error {
```

Comment:
Steffen added functionality for calculating the metrics when they are fetched https://github.com/elastic/beats/pull/3510, and this PR is updated to use this functionality.